### ISSUE TITLE
[Bug]: Load failed with error `can't change the load field list for loaded collection` after upgrading from v2.3.15 to 2.4-20240903-df8d1c7c-amd64

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#35959]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.3.15 to 2.4-20240903-df8d1c7c-amd64
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```

[2024-09-03T07:13:16.457Z] [2024-09-03 07:12:04 - INFO - ci_test]: assert flush: 10.201355218887329, entities: 76358 (test_all_collections_after_chaos.py:67)

[2024-09-03T07:13:16.457Z] [2024-09-03 07:12:04 - INFO - ci_test]: index info: [{'collection': 'Checker__w1oZAyLO', 'field': 'float_vector', 'index_name': 'vec_index', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:71)

[2024-09-03T07:13:16.458Z] [2024-09-03 07:12:04 - INFO - ci_test]: index info: [{'collection': 'Checker__w1oZAyLO', 'field': 'float_vector', 'index_name': 'vec_index', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:86)

[2024-09-03T07:13:16.458Z] [2024-09-03 07:12:04 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-09-03T07:13:16.458Z] [2024-09-03 07:12:04 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=65535, message=call query coordinator LoadCollection: can't change the load field list for loaded collection: invalid parameter[expected=[]][actual=[100 101 102 103 104]])>, <Time:{'RPC start': '2024-09-03 07:12:04.484400', 'RPC error': '2024-09-03 07:12:04.487397'}> (decorators.py:147)

[2024-09-03T07:13:16.458Z] [2024-09-03 07:12:04 - ERROR - ci_test]: Traceback (most recent call last):

[2024-09-03T07:13:16.458Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-09-03T07:13:16.458Z]     res = func(*args, **_kwargs)

[2024-09-03T07:13:16.458Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-09-03T07:13:16.458Z]     return func(*arg, **kwargs)

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 425, in load

[2024-09-03T07:13:16.458Z]     conn.load_collection(

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 148, in handler

[2024-09-03T07:13:16.458Z]     raise e from e

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 144, in handler

[2024-09-03T07:13:16.458Z]     return func(*args, **kwargs)

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 183, in handler

[2024-09-03T07:13:16.458Z]     return func(self, *args, **kwargs)

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 123, in handler

[2024-09-03T07:13:16.458Z]     raise e from e

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 87, in handler

[2024-09-03T07:13:16.458Z]     return func(*args, **kwargs)

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 1160, in load_collection

[2024-09-03T07:13:16.458Z]     check_status(response)

[2024-09-03T07:13:16.458Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 63, in check_status

[2024-09-03T07:13:16.458Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-09-03T07:13:16.458Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=65535, message=call query coordinator LoadCollection: can't change the load field list for loaded collection: invalid parameter[expected=[]][actual=[100 101 102 103 104]])>

[2024-09-03T07:13:16.458Z]  (api_request.py:45)
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_kafka_for_release_cron/detail/deploy_test_kafka_for_release_cron/2840/pipeline
log:

[artifacts-kafka-cluster-upgrade-2840-server-second-deployment-logs.tar.gz](https://github.com/user-attachments/files/16861755/artifacts-kafka-cluster-upgrade-2840-server-second-deployment-logs.tar.gz)

**Comment by zhuwenxing** [comment]
This issue has been coming up since yesterday, possibly as a result of the commit of the most recent day.
@congqixia 
PTAL

**Comment by zhuwenxing** [comment]
also failed when upgrading from v2.3.15 to master-20240903-f68df9a1-amd64

**Comment by zhuwenxing** [comment]
verified and fixed in `2.4-20240908-835c9d5c-amd64`
